habitat/tasks/nav/nav_task.py

```diff
@@ -363,6 +443,148 @@ def update_metric(self, episode, action):
             self._metric += 1
 
 
+class TopDownMap(habitat.Measure):
+    """Top Down Map measure
+    """
+
+    def __init__(self, sim: Simulator, config: Config):
+        self._sim = sim
+        self._draw_source_and_target = config.DRAW_SOURCE_AND_TARGET
+        self._draw_border = config.DRAW_BORDER
+        self._grid_delta = 3
+        self._step_count = 0
+        self._max_steps = config.MAX_EPISODE_STEPS
+        self._map_resolution = (config.MAP_RESOLUTION, config.MAP_RESOLUTION)
+        self._num_samples = 20000
+        self._ind_x_min = None
+        self._ind_x_max = None
+        self._ind_y_min = None
+        self._ind_y_max = None
+        self._previous_xy_location = None
+        self._coordinate_min = maps.COORDINATE_MIN
+        self._coordinate_max = maps.COORDINATE_MAX
+        self._top_down_map = None
+        self._cell_scale = (
+            self._coordinate_max - self._coordinate_min
+        ) / self._map_resolution[0]
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "top_down_map"
+
+    def _check_valid_nav_point(self, point: List[float]):
+        return (
+            0.01
+            < self._sim.geodesic_distance(
+                point, [point[0], point[1] + 0.1, point[2]]
+            )
+            < 0.11
```

Comment:
Use class parameters for constants.

Reply:
@erikwijmans Was there a new habitat-sim function that we are supposed to use now for checking whether a point is valid or not?

Reply:
Yep, https://github.com/facebookresearch/habitat-api/blob/master/habitat/sims/habitat_simulator.py#L341